File: ClosestNumber.c
#include <stdio.h>
#include <stdlib.h>
int closestNumber(int n, int m) {
    if (n % m == 0) {
        return n;
    } else {
        int value = n / m;
        // Tính số nguyên chia hết nhỏ hơn hoặc bằng n (leftNum)
        int leftNum = m * value;
        // Khai báo biến rightNum cho số nguyên chia hết lớn hơn n
        int rightNum;
        // Nếu leftNum là số dương, tính rightNum lớn hơn leftNum
        if (leftNum > 0) {
            rightNum = m * (value + 1); // Tính rightNum lớn hơn
        } else {
            // Nếu leftNum là số âm, tính rightNum nhỏ hơn leftNum
            rightNum = m * (value - 1); // Tính rightNum nhỏ hơn
        }
        if (abs(n - leftNum) < abs(n - rightNum)) {
            return leftNum; // Nếu leftNum gần n hơn, trả về leftNum
        } else {
            return rightNum; // Nếu rightNum gần n hơn, trả về rightNum
        }
    }
}

int main() {
    int n, m;

    // Nhập hai số nguyên n và m
    printf("Hãy nhập 2 số nguyên bất kì cách nhau bởi khoảng trắng: ");
    scanf("%d %d", &n, &m);

    // Gọi hàm closestNumber để tìm số gần n nhất chia hết cho m
    int result = closestNumber(n, m);

    // In kết quả ra màn hình
    printf("\nSố nguyên gần %d nhất chia hết cho %d là: %d\n", n, m, result);

    return 0;
}

File: ClosestNumber.py
def closestNumber(n, m) :
    q = int(n // m)
    n1 = m * q
    # Số gần nhất thứ hai có thể chia hết cho m (n2)
    if((n * m) > 0) :
        # Nếu n và m có cùng dấu (cả hai đều dương hoặc đều âm),ta lấy số chia hết cho m lớn hơn n1
        n2 = (m * (q + 1)) 
    else :
        # Nếu n và m có dấu khác nhau, ta lấy số chia hết cho m nhỏ hơn n1
        n2 = (m * (q - 1))
    # abs(n - n1) là khoảng cách giữa n và n1
    # abs(n - n2) là khoảng cách giữa n và n2
    if (abs(n - n1) < abs(n - n2)) :
        return n1
    return n2

# Nhập vào hai số n và m từ bàn phím
m, n = map(int, input("Hãy nhập hai số nguyên bất kì: ").split())

# In ra kết quả số gần nhất với m và chia hết cho n
print(f"Số nguyên gần {m} nhất chia hết cho {n} là: {closestNumber(m, n)}")
